src/main/java/org/skriptlang/skript/registration/SyntaxRegister.java

```diff
@@ -14,13 +15,44 @@
  */
 final class SyntaxRegister<I extends SyntaxInfo<?>> {
 
+	private static int calculateComplexityScore(SyntaxInfo<?> info) {
+		return info.patterns().stream()
+				.mapToInt(SyntaxRegister::calculateComplexityScore)
+				.max()
+				.orElseThrow(); // a syntax info should have at least one pattern
+	}
+
+	private static int calculateComplexityScore(String pattern) {
+		int score = 0;
+		char[] chars = pattern.toCharArray();
+		for (int i = 0; i < chars.length; i++) {
+			if (chars[i] == '%') {
+				if (i - 2 >= 0 && chars[i - 2] == '%') { // weigh "%thing% %thing%" heavier
+					score += 3;
+				} else {
+					score++;
+				}
+			}
+		}
+		return score;
+	}
+
 	private static final Comparator<SyntaxInfo<?>> SET_COMPARATOR = (a,b) -> {
 		if (a == b) { // only considered equal if registering the same infos
 			return 0;
 		}
-		int result = a.priority().compareTo(b.priority());
-		// when elements have the same priority, order by hashcode
-		return result != 0 ? result : Integer.compare(a.hashCode(), b.hashCode());
+		// priority is the primary factor in determining ordering
+		int priorityResult = a.priority().compareTo(b.priority());
+		if (priorityResult != 0) {
+			return priorityResult;
+		}
+		// otherwise, consider the complexity of the syntax
+		int scoreResult = Integer.compare(calculateComplexityScore(a), calculateComplexityScore(b));
```

Comment:
Should we be caching this score? Seems like it might be calculated a lot.

Reply:
Potentially. This would mainly be called at first startup and potentially during a script reload, so I wasn't sure how much it really mattered.

Reply:
Sure, but that comparison could be called thousands of times for a few hundred syntaxes, so I think it's probably worth it.

Reply:
As discussed on Discord, we'd have to decide on which is going to be more expensive: this call repeatedly occurring (which is only a problem during startup and rarely script reloading) OR to rework the structure so that the priority is cached. Putting it on SyntaxInfo directly is likely not feasible (for now), so we would have to store it as a record in the set, which would require mapping back to a SyntaxInfo for iteration (slightly more expensive for all reloads)